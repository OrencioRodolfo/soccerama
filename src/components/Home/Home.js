import React from 'react'
import styled from 'styled-components'
import logo from '../../images/logo.png'
import Link from '../Link/Link'
import League from '../League/League'

const Page = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  height: 100%;
  padding: 10px;
  box-sizing: border-box;
  max-width: 1024px;
`

const Head = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
`
const Img = styled.img`
  width: 100%;
  min-width: 120px;
  max-width: 500px;
  height: auto;
`

const Content = styled.div`
  flex-grow: 1;
  flex-shrink: 0;
  padding: 30px 0;
  overflow-x: scroll;
`

const Footer = styled.div`
  border-top: 1px solid grey;
  padding: 10px 0;
`

const url = 'https://www.sportmonks.com'

const Home = () => (
  <Page>
    <Head>
      <Link href={url}>
        <Img src={logo} alt="soccreama" />
      </Link>
    </Head>
    <Content>
      <League />
    </Content>
    <Footer>
      <Link href={url}>SportMonks</Link> provides fast and reliable data
      for different kind of sports.
    </Footer>
  </Page>
)

export default Home
