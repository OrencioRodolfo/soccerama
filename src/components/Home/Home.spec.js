import React from 'react'
import { shallow } from 'enzyme'
import Home from './Home'

describe('Home component', () => {
  it('must render properly', () => {
    const component = shallow(<Home />)

    expect(component).toMatchSnapshot()
  })
})
