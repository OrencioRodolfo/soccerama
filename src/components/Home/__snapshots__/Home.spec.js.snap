// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <Link
            href="https://www.sportmonks.com"
          >
            <styled.img
              alt="soccreama"
              src="test-file-stub"
            />
          </Link>
        </styled.div>,
        <styled.div>
          <Connect(League) />
        </styled.div>,
        <styled.div>
          <Link
            href="https://www.sportmonks.com"
          >
            SportMonks
          </Link>
           provides fast and reliable data for different kind of sports.
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            href="https://www.sportmonks.com"
          >
            <styled.img
              alt="soccreama"
              src="test-file-stub"
            />
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <styled.img
              alt="soccreama"
              src="test-file-stub"
            />,
            "href": "https://www.sportmonks.com",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "soccreama",
              "src": "test-file-stub",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(League) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              href="https://www.sportmonks.com"
            >
              SportMonks
            </Link>,
            " provides fast and reliable data for different kind of sports.",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "SportMonks",
              "href": "https://www.sportmonks.com",
            },
            "ref": null,
            "rendered": "SportMonks",
            "type": [Function],
          },
          " provides fast and reliable data for different kind of sports.",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <Link
              href="https://www.sportmonks.com"
            >
              <styled.img
                alt="soccreama"
                src="test-file-stub"
              />
            </Link>
          </styled.div>,
          <styled.div>
            <Connect(League) />
          </styled.div>,
          <styled.div>
            <Link
              href="https://www.sportmonks.com"
            >
              SportMonks
            </Link>
             provides fast and reliable data for different kind of sports.
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              href="https://www.sportmonks.com"
            >
              <styled.img
                alt="soccreama"
                src="test-file-stub"
              />
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <styled.img
                alt="soccreama"
                src="test-file-stub"
              />,
              "href": "https://www.sportmonks.com",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "soccreama",
                "src": "test-file-stub",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(League) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                href="https://www.sportmonks.com"
              >
                SportMonks
              </Link>,
              " provides fast and reliable data for different kind of sports.",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "SportMonks",
                "href": "https://www.sportmonks.com",
              },
              "ref": null,
              "rendered": "SportMonks",
              "type": [Function],
            },
            " provides fast and reliable data for different kind of sports.",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
